.travis.yml

```diff
@@ -1,25 +1,48 @@
-language: python
-services:
+language: bash
+sudo: required
+service:
 - docker
+install:
+- curl -sSL -o terraform_"$TF_VERSION"_linux_amd64.zip "https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip"
+#- sudo wget https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip
+- sudo unzip -o terraform_"$TF_VERSION"_linux_amd64.zip
+#- sudo unzip -f terraform.zip
+- sudo mv terraform /usr/local/bin/
+- rm terraform_"$TF_VERSION"_linux_amd64.zip
+#- rm terraform.zip
 jobs:
   include:
-  - stage: build and test
-    script:
-    - docker build --target test  --tag todo-app:test .
-    - docker run -e SECRET_KEY -e API_KEY -e API_TOKEN todo-app:test tests_e2e
-    - docker run --env-file ./.env.test todo-app:test tests
-
-  - stage: docker build and publish
-    script:
-    - docker build --target production --tag $DOCKER_USERNAME/todo-app:prod .
-    - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-    - docker push $DOCKER_USERNAME/todo-app:prod
-
-deploy:
-  provider: script
-  script: bash ./deploy.sh
-  on:
-    branch: main
+    - stage: build and test
+      script:
+      - cd terraform_env/dev
+      - terraform init
+      - terraform plan -var "prefix=test" -var "location=$LOCATION" -var "resource_group_name=$RESOURCE_GROUP_NAME"
+      - terraform apply -var "prefix=test" -var "location=$LOCATION" -var "resource_group_name=$RESOURCE_GROUP_NAME" -auto-approve
```

Comment:
If you want to run `terraform plan` immediately followed by `terraform apply`, use `-out=FILE` to actually save your plan to a file and then pass that to the apply step. That way it is actually guaranteed to do what the plan step said it would.